### PR TITLE
[docs] Restructure explanations of test types

### LINKED ISSUE #5: Downplay relevance of visual tests
People looking to learn how to write a test for WPT have a lot to learn.

The "Writing Tests" page lists four different test types, including both "Visual Tests" and "Reftests". Much of the guidance for these tests is located in a more generic document, "Rendering Test Guidelines."

- Writing Tests
  - Reftests
    - Rendering Test Guidelines
  - Visual Tests
    - Rendering Test Guidelines

Since "Visual Tests" are so uncommon (WPT currently has 35 in total) and since the guidelines for writing them is so minimal (six sentences), I think they ought to be demoted in the site hierarchy. By documenting them instead as a special case of Rendering Tests, we can meaningfully decrease the number of concepts which potential contributors are initially confronted with.

- Writing Tests
  - Rendering Tests (+Guidelines)
    - Reftests
    - Visual Tests

This also demotes the far more pervasive Reftest. I think that's acceptable because the path is still clear, but we might also choose to merge "Rendering Tests" and "Reftests", and just document Visual Tests as a special case:

- Writing Tests
  - Rendering Tests (+Guidelines, +Reftests)
    - Visual Tests

Neither of these structures directly map to the organization of wptrunner internals, but I think the disconnect is acceptable if it meaningfully improves the documentation's usability.

### PR DESCRIPTION
**Pull request by jugglinmike**
The project documentation includes two lists of WPT's various test
types. Update these lists in the interest of helping readers orient
themselves to the project.

- Present reftests and visual tests under a generic "rendering test"
  item in recognition of the related nature of those two test types. Use
  the new item in the "Writing Tests" document to recommend the
  rendering test guidelines.
- Consistently introduce WebDriver tests in both lists

---

This is an attempt to resolve [bocoup/wpt-docs#5](https://github.com/bocoup/wpt-docs/issues/5). It's a little less ambitious than the plan outlined there because it only changes the structure of the table of contents--not the documents themselves. This doesn't reduce the number of concepts we present, but I'm hoping it still helps readers build an understanding.

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

**Review by Ms2ger — APPROVED** [review]

File: docs/_writing-tests/index.md [hunk 1]
```diff
@@ -11,43 +11,47 @@ There's also a load of [general guidelines][] that apply to all tests.
...
+  * [Reftests][] should be used to test rendering and layout. They consist of
+    two or more pages with assertions as to whether they render identically or
+    not.
 
-* [Visual tests][visual] should be used for checking rendering where
-  there is a large number of conforming renderings such that reftests
-  are impractical. They consist of a page that renders to final state
-  at which point a screenshot can be taken and compared to an expected
-  rendering for that user agent on that platform.
+  * [Visual tests][visual] should be used for checking rendering where there is
```
Comment (gsnedders) [inline docs/_writing-tests/index.md]: At this point, there's only marginal value in having visual tests (they're practically identical to manual tests, given nobody has automation to handle them nowadays). We probably don't want to talk about screenshots here, but rather note they're essentially a subset of manual tests.

File: docs/_writing-tests/index.md [hunk 2]
```diff
@@ -11,43 +11,47 @@ There's also a load of [general guidelines][] that apply to all tests.
...
+* Rendering tests should be used to verify that the browser graphically
+  displays pages as expected. See the [rendering test guidelines][rendering]
+  for tips on how to write great rendering tests. There are a few different
+  ways to write rendering tests:
 
-* [testharness.js][] tests should be used (where
-  possible!) for testing everything else. They are built with the
-  testharness.js unit testing framework, and consist of assertions
-  written in JavaScript.
+  * [Reftests][] should be used to test rendering and layout. They consist of
```
Comment (gsnedders) [inline docs/_writing-tests/index.md]: Note that this is increasingly not the case; more and more of our layout tests are testharness.js tests. Changing the documentation may be more contentious however.

File: docs/_writing-tests/index.md [hunk 3]
```diff
@@ -11,43 +11,47 @@ There's also a load of [general guidelines][] that apply to all tests.
...
-* [Manual tests][manual] are used as a last resort for anything
-  that can't be tested using any of the above. They consist of a page
-  that needs manual interaction or verification of the final result.
+* [testharness.js][] tests should be used (where possible!) for testing
+  everything else. They are built with the testharness.js unit testing
+  framework, and consist of assertions written in JavaScript.
 
-In general, there is a strong preference towards the first two test
-types (as they can be easily run without human interaction), so they
+* WebDriver tests are written in Python and test [the WebDriver browser
```
Comment (gsnedders) [inline docs/_writing-tests/index.md]:
```suggestion
* WebDriver tests are written in Python using `pytest` and test [the WebDriver browser
```

(probably with a link)

**Review by gsnedders — COMMENTED** [review]
The duplication here is somewhat annoying :(

File: docs/_writing-tests/index.md (hunk 1 above)
Comment (jugglinmike) [reply docs/_writing-tests/index.md]: It sounds like Visual Tests can be removed from these overviews completely, and a reference instead added to the "Manual Tests" page. Would you agree?

File: docs/_writing-tests/index.md (hunk 2 above)
Comment (jugglinmike) [reply docs/_writing-tests/index.md]: More contentious, and maybe more involved as well. We might want to make some different/more specific recommendations for how authors should choose between the types. Let's discuss that in a follow-up.

File: docs/_writing-tests/index.md (hunk 3 above)
Comment (jugglinmike) [reply docs/_writing-tests/index.md]: Good idea!

File: docs/_writing-tests/index.md (hunk 1 above)
Comment (gsnedders) [reply docs/_writing-tests/index.md]: So I have some concern about it becoming hard to find out what `-visual` tests are (especially once we bulk rename CSS tests in #5381), even though really they're just a special case of manual tests at this point.

**Review by sideshowbarker — APPROVED** [review]

File: docs/_writing-tests/index.md (hunk 2 above)
Comment (gsnedders) [reply docs/_writing-tests/index.md]: @jugglinmike can you file an issue for this?

Comment (jugglinmike) [reply docs/_writing-tests/index.md]: Sure: gh-16602

**Review by gsnedders — APPROVED** [review]

**Pull request merged**